website/source/docs/secrets/pki/index.html.md

```diff
@@ -0,0 +1,686 @@
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">common_name</span>
+        <span class="param-flags">required</span>
+        The requested CN for the certificate. If the CN is allowed
+        by role policy, it will be issued.
+      </li>
+      <li>
+        <span class="param">alt_names</span>
+        <span class="param-flags">optional</span>
+        Requested Subject Alternative Names, in a comma-delimited
+        list. If any requested names do not match role policy,
+        the entire request will be denied.
+      </li>
+      <li>
+        <span class="param">ip_sans</span>
+        <span class="param-flags">optional</span>
+        Requested IP Subject Alternative Names, in a comma-delimited
+        list. Only valid if the role allows IP SANs (which is the
+        default).
+      </li>
+      <li>
+      <span class="param">lease</span>
+      <span class="param-flags">optional</span>
+        Requested lease time. Cannot be greater than the role's
+        `lease_max` parameter. If not provided, the role's `lease`
+        value will be used.
+      </li>
+    </ul>
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+
+  ```javascript
+  {
+      "lease_id": "pki/issue/test/7ad6cfa5-f04f-c62a-d477-f33210475d05",
+      "renewable": false,
+      "lease_duration": 21600,
+      "data": {
+          "certificate": "-----BEGIN CERTIFICATE-----\nMIIDzDCCAragAwIBAgIUOd0ukLcjH43TfTHFG9qE0FtlMVgwCwYJKoZIhvcNAQEL\n...\numkqeYeO30g1uYvDuWLXVA==\n-----END CERTIFICATE-----\n",
+          "issuing_ca": "-----BEGIN CERTIFICATE-----\nMIIDUTCCAjmgAwIBAgIJAKM+z4MSfw2mMA0GCSqGSIb3DQEBCwUAMBsxGTAXBgNV\n...\nG/7g4koczXLoUM3OQXd5Aq2cs4SS1vODrYmgbioFsQ3eDHd1fg==\n-----END CERTIFICATE-----\n",
+          "private_key": "-----BEGIN RSA PRIVATE KEY-----\nMIIEowIBAAKCAQEAnVHfwoKsUG1GDVyWB1AFroaKl2ImMBO8EnvGLRrmobIkQvh+\n...\nQN351pgTphi6nlCkGPzkDuwvtxSxiCWXQcaxrHAL7MiJpPzkIBq1\n-----END RSA PRIVATE KEY-----\n",
+          "serial": "39:dd:2e:90:b7:23:1f:8d:d3:7d:31:c5:1b:da:84:d0:5b:65:31:58"
+      },
+      "auth": null
+  }
+  ```
+
+  </dd>
+</dl>
+
+### /pki/revoke
+#### POST
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Revokes a certificate using its serial number. This is an
+    alternative option to the standard method of revoking
+    using Vault lease IDs. A successful revocation will
+    rotate the CRL.
+    <br /><br />This is a root-protected endpoint.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>POST</dd>
+
+  <dt>URL</dt>
+  <dd>`/pki/revoke`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">serial</span>
+        <span class="param-flags">required</span>
+        The serial number of the certificate to revoke, in
+        hyphen-separated or colon-separated octal.
+      </li>
+    </ul>
+  </dd>
+
+  <dt>Returns</dt>
+  <dd>
+  ```javascript
+  {
+      "data": {
+          "revocation_time": 1433269787
+      }
+  }
+  ```
+  </dd>
+</dl>
+
+### /pki/revoked/
```

Comment:
It gives you two things: the revoked certificate and the revocation time. This could potentially be useful for auditing/forensics.

I'm not sure what to tell you about the strangeness. It makes sense to us from a completeness standpoint.

If you don't want `revoked/`, the question then is whether to make revoked certificates accessible at all.